immutablecollections/_immutabledict.py

```diff
@@ -19,21 +20,23 @@
 
 KT = TypeVar("KT")
 VT = TypeVar("VT")
-IT = Tuple[KT, VT]
 
 # cannot share type variables between outer and inner classes
 KT2 = TypeVar("KT2")
 VT2 = TypeVar("VT2")
-IT2 = Tuple[KT2, VT2]
 
 SelfType = TypeVar("SelfType")  # pylint:disable=invalid-name
 
-AllowableSourceType = Union[Iterable[IT], Mapping[KT, VT], "ImmutableDict[KT, VT]"]
+AllowableSourceType = Union[
+    Iterable[Tuple[KT, VT]], Mapping[KT, VT], "ImmutableDict[KT, VT]"
+]
 InstantiationTypes = (Mapping, Iterable)  # pylint:disable=invalid-name
 
 
 def immutabledict(
-    iterable: Optional[AllowableSourceType] = None, *, forbid_duplicate_keys: bool = False
+    iterable: Optional[AllowableSourceType[Any, Any]] = None,
```

Comment:
What happens if you use `KT, VT` instead of `Any, Any`?

Reply:
It throws errors on `vistautils` saying that typing doesn't match. For some reason it tries to match to `KT` and `VT` exactly instead of using them as `typevar`s. This can be seen as the original fix committed used `typevar`s directly which `mypy` was okay with despite it being a syntax error on the typing. Furthermore, the errors thrown by mypy when using `KT, VT` were not errors from the flags so everyone who uses `immutablecollections` would be affected even if we just `# type: ignore`ed on `vistautils`.